src/documents/firstDiscoveryServer.md

```diff
@@ -0,0 +1,344 @@
+            <td>
+                "8081"
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <code>"preferencesConfig.securityServer.hostname"</code>
+            </td>
+            <td>
+                The hostname that the GPII OAuth2 server is hosted at.
+            </td>
+            <td>
+                "http://10.0.2.2"
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+## Launching ##
+
+The First Discovery Server can be launched as a [Kettle](https://github.com/amb26/kettle/blob/KETTLE-32/README.md) application by making use of [Kettle Configs](https://github.com/amb26/kettle/blob/KETTLE-32/README.md#structure-of-a-kettle-config).
+
+There are two typical ways of launching a Kettle app, programmatically and from command line.
+
+(See: [Starting a Kettle application](https://github.com/amb26/kettle/blob/KETTLE-32/README.md#starting-a-kettle-application))
+
+### Programmatically ###
+
+```javascript
+// require the kettle module
+var kettle = require("kettle");
+
+// load the config
+kettle.config.loadConfig({
+    // path to the config directory
+    configPath:"./src/config",
+
+    // name of the config to load, without the file extension
+    configName: "vagrant"
+});
+```
+
+### Command Line ####
+
+```bash
+# Call Kettle's init.js script with the
+# configPath and configName
+# node node_modules/kettle/init.js <configPath> [<configName>]
+node node_modules/kettle/init.js ./src/config vagrant
+
+# or using an environment variable to specify
+# the configName
+# NODE_EVN=<configName> node node_modules/kettle/init.js <configPath>
+NODE_EVN=vagrant node node_modules/kettle/init.js ./src/config
+```
+
+## REST API ##
+
+<table>
+    <thead>
+        <tr>
+            <th>URL</th>
+            <th>Request</th>
+            <th>Description</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td><code>/user[?view=:view]</code></td>
+            <td><code>POST</code></td>
+            <td>
+                Accepts a set of preferences, in a JSON object, to be stored on the preferences server. For example:
+
+                <pre>
+                    <code>
+{
+  "gpii_firstDiscovery_language": "en-US"
+}
+                    </code>
+                </pre>
+
+                On successfully storing the preferences set, a JSON object is returned containing `userToken` and `preferences` properties. The `userToken` is the GPII token, which can be used for retrieving the preferences on a GPII enabled device. The `preferences` are a confirmation of the preferences stored on Preferences server for that GPII token. For example:
+
+                <pre>
+                    <code>
+{
+  "userToken": "2288e676-d0bb-4d29-8131-7cff268ba012",
+  "preferences": {
+    "contexts": {
+      "gpii-default": {
+        "name": "Default preferences",
+        "preferences": {
+          "gpii_firstDiscovery_language": "en-US"
+        }
+      }
+    }
+  }
+}
+                    </code>
```

Comment:
This code block is rendering fine now, but there's an extra line at the beginning and end. Consider putting the opening and closing {} on the same line as the opening and closing code tags.